cpp/src/io/avro/reader_impl.cu

```diff
@@ -425,40 +439,26 @@ table_with_metadata reader::impl::read(avro_reader_options const &options, cudaS
             dict_pos += len;
```

Comment:
If `global_dictionary.host_ptr() + dict_pos` is indeed correct, then you can use the offset argument `global_dictionary.host_ptr(dict_pos)`